src/components/Match/VisionPage.jsx

```diff
@@ -0,0 +1,231 @@
+          </Row>
+        </Col>
+        <Col xs={12} sm={5}>
+          <Row>
+            <Col xs>
+              <Button {...this.getMuiThemeProps()}
+                      label={obs_count}
+                      disabled={obs_count == 0}
+                      backgroundColor={this.generateFilterKey("observer") in this.props.activeFilters ? filterOff : filterOn}
+                      style={{opacity: obs_count > 0 ? opacityOn : opacityOff}}
+                      onClick={() => onFilterClick(this.generateFilterKey("observer"), player.player_slot, "observer")}
+                      icon={<Avatar size={24} src="http://a19a1164.ngrok.io/apps/dota2/images/items/ward_observer_lg.png" />}
+              />
+            </Col>
+            <Col xs>
+              <Button {...this.getMuiThemeProps()}
+                      label={sen_count}
+                      disabled={sen_count == 0}
+                      backgroundColor={this.generateFilterKey("sentry") in this.props.activeFilters ? filterOff : filterOn}
+                      style={{opacity: sen_count > 0 ? opacityOn : opacityOff}}
+                      onClick={() => onFilterClick(this.generateFilterKey("sentry"), player.player_slot, "sentry")}
+                      icon={<Avatar size={24} src="http://a19a1164.ngrok.io/apps/dota2/images/items/ward_sentry_lg.png" />}
+              />
+            </Col>
+          </Row>
+        </Col>
+      </Row>
+    );
+  }
+}
+
+const PlayersFilter = ({ activeFilters, players, onFilterClick }) => (
+  <Paper className={styles['ward-log-player-filter']}>
+    {players.map((p,i) => <PlayerFilter player={p} activeFilters={activeFilters} onFilterClick={onFilterClick} />)}
+  </Paper>
+);
+
+import Immutable from 'seamless-immutable';
+const PipelineFilter = (filters, data, iter=Array.prototype.filter) => {
+  const frozenData = Immutable(data);
+  let filtered = filters.map(f => iter.call(frozenData, f))
+                        .reduce((o, v) => o.concat(v), []);
+  return _.differenceWith((x, y) => x === y, frozenData, filtered);
+}
+
+const FixedPlayersFilter = PlayersFilter;
+
+class VisionPage extends React.Component {
+  componentWillMount() {
+    Perf.start();
+  }  
+  
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentTick: -90,
+      min: -90,
+      max: props.match.duration,
+      from: 0,
+      to: this.props.match.wards_log.length,
+      wardsLog: props.match.wards_log.map((e, i) => Object.assign(e, {key: i})),
+      filters: {},
+    }
+
+    this.ticks = this.computeTick();
+    this.findPivot = (value) => _.flow(_.map(x => x.entered.time),
+                                       _.sortedIndex(value))(this.state.wardsLog);
+    this.handleViewportChange = _.debounce(50, this._handleViewportChange);
+  }
+
+  computeTick() {
+    const interval = 10 * 60; // every 10 minutes interval
+    return _.rangeStep(interval, 0, this.props.match.duration);
+  }
+
+  _handleViewportChange(value) {
```

Comment:
I think eslint will complain about the use of underscores since javascript doesn't really have any idea of private properties.